kafka-streams-quickstart/README.md

```diff
@@ -0,0 +1,166 @@
+```
+
+## Scaling
+
+Kafka Streams pipelines can be scaled out, i.e. the load can be distributed amongst multiple application instances running the same pipeline.
+To try this out, scale the _aggregator_ service to three nodes:
+
+```bash
+docker-compose up -d --scale aggregator=3
+```
+
+This will spin up two more instances of this service.
+The state store that materializes the current state of the streaming pipeline
+(which we queried before using the interactive query),
+is now distributed amongst the three nodes.
+I.e. when invoking the REST API on any of the three instances, it might either be
+that the aggregation for the requested weather station id is stored locally on the node receiving the query,
+or it could be stored on one of the other two nodes.
+
+As the load balancer of Docker Compose will distribute requests to the _aggregator_ service in a round-robin fashion,
+we'll invoke the actual nodes directly.
+The application exposes information about all the host names via REST:
+
+```bash
+http aggregator:8080/weather-stations/meta-data
+```
+
+Retrieve the data from one of the three hosts shown in the response
+(your actual host names will differ):
+
+```bash
+http cf143d359acc:8080/weather-stations/data/1
+```
+
+If that node holds the data for key "1", you'll get a response like this:
+
+```
+HTTP/1.1 200 OK
+Connection: keep-alive
+Content-Length: 74
+Content-Type: application/json
+Date: Tue, 11 Jun 2019 19:16:31 GMT
+
+{
+    "id": 1,
+    "max": 49.96072870492197,
+    "min": 0.05763795979973341,
+    "name": "Boston"
+}
+```
+
+Otherwise, the service will send a redirect:
+
+```
+HTTP/1.1 303 See Other
+Connection: keep-alive
+Content-Length: 0
+Date: Tue, 11 Jun 2019 19:17:51 GMT
+Location: http://72064bb97be9:8080/weather-stations/data/2
+```
+
+You can have _httpie_ automatically follow the redirect by passing the `--follow option`:
+
+```bash
+http --follow cf143d359acc:8080/weather-stations/data/2
+```
+
+## Running in native
+
+To run the _producer_ and _aggregator_ applications as native binaries via GraalVM,
+first run the Maven builds using the `native` profile:
+
+```bash
+mvn clean package -f producer/pom.xml -Pnative -Dnative-image.container-runtime=docker
+mvn clean package -f aggregator/pom.xml -Pnative -Dnative-image.container-runtime=docker
+```
+
+Then create an environment variable named `QUARKUS_MODE` and with value set to "native":
+
+```bash
+export QUARKUS_MODE=native
+```
+
+Now start Docker Compose as described above.
+
+## Running locally
+
+For development purposes it can be handy to run the _producer_ and _aggregator_ applications
+directly on your local machine instead of via Docker.
+For that purpose, a separate Docker Compose file is provided which just starts Apache Kafka and ZooKeeper,
+configured to be accessible from your host system:
+
+```bash
+docker-compose -f docker-compose-local.yaml up
+
+mvn compile quarkus:dev -f producer/pom.xml
+mvn compile quarkus:dev -Dquarkus.http.port=8081 \
```

Comment:
The command fails when `HOSTNAME` is not set

Reply:
Added.